src/test/scala/org/ergoplatform/utils/ErgoNodeViewHolderTestHelpers.scala

```diff
@@ -0,0 +1,144 @@
+        verifyTransactions = c.verifyTransactions,
+        PoPoWBootstrap = c.popowBootstrap
+      ),
+      chainSettings = defaultSettings.chainSettings.copy(powScheme = DefaultFakePowScheme)
+    )
+    ErgoNodeViewRef(settings, timeProvider, emission)
+  }
+
+
+  type H = ErgoHistory
+  type S = ErgoState[_]
+  type D = DigestState
+  type U = UtxoState
+  type W = ErgoWallet
+  type P = ErgoMemPool
+  type C = NodeViewHolderConfig
+
+  val allConfigs: List[NodeViewHolderConfig] = List(
+    NodeViewHolderConfig(StateType.Digest, verifyTransactions = true, popowBootstrap = true),
+    NodeViewHolderConfig(StateType.Digest, verifyTransactions = false, popowBootstrap = true),
+    NodeViewHolderConfig(StateType.Digest, verifyTransactions = false, popowBootstrap = false),
+    NodeViewHolderConfig(StateType.Digest, verifyTransactions = true, popowBootstrap = false),
+    NodeViewHolderConfig(StateType.Utxo, verifyTransactions = true, popowBootstrap = true),
+    NodeViewHolderConfig(StateType.Utxo, verifyTransactions = true, popowBootstrap = false),
+  )
+
+  protected def checkAfterGenesisState(c: C) = GetDataFromCurrentView[H, S, W, P, Boolean] { v =>
+    java.util.Arrays.equals(v.state.rootHash, settings.chainSettings.monetary.afterGenesisStateDigest)
+  }
+
+  protected def bestHeaderOpt(c: C) = GetDataFromCurrentView[H, S, W, P, Option[Header]](v => v.history.bestHeaderOpt)
+
+  protected def historyHeight(c: C) = GetDataFromCurrentView[H, S, W, P, Int](v => v.history.headersHeight)
+
+  protected def heightOf(id: ModifierId, c: C) = GetDataFromCurrentView[H, S, W, P, Option[Int]](v => v.history.heightOf(id))
+
+  protected def lastHeadersLength(count: Int, c: C) =
+    GetDataFromCurrentView[H, S, W, P, Int](v => v.history.lastHeaders(count).size)
+
+
+  protected def openSurfaces(c: C) = GetDataFromCurrentView[H, S, W, P, Seq[ModifierId]] { v =>
+    v.history.openSurfaceIds()
+  }
+
+  protected def bestFullBlock(c: C) = GetDataFromCurrentView[H, S, W, P, Option[ErgoFullBlock]] { v =>
+    v.history.bestFullBlockOpt
+  }
+
+  protected def modifierById(id: ModifierId) = GetDataFromCurrentView[H, S, W, P, Option[ErgoPersistentModifier]] { v =>
+    v.history.modifierById(id)
+  }
+
+  protected def bestFullBlockEncodedId(c: C) = GetDataFromCurrentView[H, S, W, P, Option[String]] { v =>
+    v.history.bestFullBlockOpt.map(_.header.encodedId)
+  }
+
+  protected def poolSize(c: C) = GetDataFromCurrentView[H, S, W, P, Int](v => v.pool.size)
+
+  protected def rootHash(c: C) = GetDataFromCurrentView[H, S, W, P, String](v => Algos.encode(v.state.rootHash))
+
+  case class TestCase(name: String)(test: NodeViewFixture => Unit) {
+    def run(c: NodeViewHolderConfig): Unit = new NodeViewFixture(c).run(test)
+  }
+
+  /** This uses TestProbe to receive messages from actor.
+    * To make TestProbe work `defaultSender` implicit should be imported
+    */
+  class NodeViewFixture(val nodeViewConfig: NodeViewHolderConfig) {
```

Comment:
I think it's better to move this class to outer level, as while it's nested we will have different `NodeViewFixture` classes for every `ErgoNodeViewHolderTestHelpers` descendant